force-app/main/default/classes/QueueablePokemonJob.cls

```diff
@@ -0,0 +1,152 @@
+public class QueueablePokemonJob implements Queueable, Database.AllowsCallouts {
+  public static Integer batchSize = 5;
+  private List<PokeAPICallout.PokemonData> pokemonDataList;
+  public static final String POKE_URL = 'https://pokeapi.co/api/v2/pokemon?';
+  private Integer batchNum;
+  private Integer pokemonLeft;
+  private static Integer numPokemon;
+  private Integer batch;
+  private static Id accountId;
+  private List<Pokemon__c> pokemonList = new List<Pokemon__c>();
+
+  public QueueablePokemonJob(
+    List<PokeAPICallout.PokemonData> pokemonDataList,
+    Integer pokemonLeft,
+    Integer batchNum,
+    Integer batch
+  ) {
+    this.pokemonDataList = pokemonDataList;
+    this.pokemonLeft = pokemonLeft;
+    this.batchNum = batchNum;
+    this.batch = batch;
+  }
+
+  public void execute(QueueableContext context) {
+    // loop through context
+    // for the first batchSize # of records, place record into different list and remove from context (pokemonDataList)
+
+    if (pokemonLeft == 0 && batchNum == 0) {
+      if (!pokemonDataList.isEmpty()) {
+        PokeAPICallout.PokemonData pokemonData = pokemonDataList[0]; // Get the first element
+        numPokemon = pokemonData.numPokemon;
+        accountId = pokemonData.accountId;
+        batch = 0;
+        batchNum = 0;
+        pokemonLeft = numPokemon;
+      }
+    } else {
+      if (!pokemonDataList.isEmpty()) {
+        PokeAPICallout.PokemonData pokemonData = pokemonDataList[0];
+        numPokemon = pokemondata.numPokemon;
+        pokemonLeft = numPokemon - (batchSize * batchNum); // Reset pokemonLeft for the current batch
+        accountId = pokemonData.accountId;
+      }
+    }
+
+    for (Integer i = 0; i < Math.min(pokemonLeft, batchSize); i++) {
```

Comment:
Starting here, I want you to refactor all of the code below into smaller, private methods.